biollante_rodan.py

```diff
@@ -0,0 +1,287 @@
+            self.logger.info("State: Not Optimizing")
+
+            # Create set of parameters for template
+            d = self.knnga_dict()
+            d["@state"] = STATE_NOT_OPTIMIZING
+            d["@settings"] = settings["@settings"]
+            return self.WAITING_FOR_INPUT(d)
+
+        elif settings["@state"] == STATE_OPTIMIZING:
+            self.logger.info("State: Optimizing")
+            self.load_from_settings(settings)
+
+            # Load data
+            with NTF(suffix=".xml") as temp:
+                shutil.copy2(
+                    inputs["kNN Training Data"][0]["resource_path"],
+                    temp.name
+                )
+                classifier = knn.kNNNonInteractive(temp.name)
+
+            # Load selection and weights
+            with NTF(suffix=".xml") as temp:
+                temp.write(settings["@settings"])
+                temp.flush()
+                classifier.load_settings(temp.name)
+
+            self.optimizer = knnga.GAOptimization(
+                classifier,
+                self.base,
+                self.selection.selection,
+                self.crossover.crossover,
+                self.mutation.mutation,
+                self.replacement.replacement,
+                self.stop_criteria.sc,
+                knnga.GAParallelization(True, 4)
+            )
+
+            assert isinstance(self.optimizer, knnga.GAOptimization), \
+                "Optimizer is %s" % str(type(self.optimizer))
+
+            try:
+                self.optimizer.startCalculation()
+            except Exception as e:
+                self.logger.error(e)
+                self.logger.error("Failed to start optimizing!")
+                return False
+
+            # Wait for optimization to finish
+            while self.optimizer.status:
+                sleep(30000)    # 30 seconds
+                self.logger.info(self.optimizer.monitorString)
+
+            # This is necessary since the classifier object isn't persistent
+            settings = self.knnga_dict()
+
+            with NTF() as temp:
+                classifier.save_settings(temp.name)
+                temp.flush()
+                temp.seek(0)
+                settings["@settings"] = temp.read()
+
+            settings["@state"] = STATE_NOT_OPTIMIZING
+            return self.WAITING_FOR_INPUT(settings)
+
+        else:   # Finish
+            self.logger.info("State: Finishing")
+            with open(
+                outputs["GA Optimized Classifier"][0]["resource_path"], 'w'
+            ) as f:
+                f.write(settings["@settings"])
+            return True
+
+    def my_error_information(self, exc, traceback):
+        raise NotImplementedError
+
+    def test_my_task(self, testcase):
+        # inputs = {}
+        # outputs = {}
+        # self.run_my_task(inputs, {}, outputs)
+        # result = outputs['out'][0]['resource_path']
+        # testcase.assertEqual(result, 'what you expect it to test')
```

Comment:
Commented out code

Reply:
@blueNoteb5 That's a placeholder on how to write a rodan job test.

Reply:
@JRegimbal you could just import and run TestSelection, TestReplacement, etc. here

Reply:
Unless we decide to remove `test_my_task` from rodan.